cnd/src/swap_protocols/hbit.rs

```diff
@@ -0,0 +1,242 @@
+pub mod events;
+mod extract_secret;
+pub mod htlc_events;
+mod ledger_state;
+mod ledger_states;
+
+pub use self::{
+    extract_secret::*,
+    htlc_events::*,
+    ledger_state::{HtlcState, LedgerState},
+    ledger_states::*,
+};
+
+use crate::{
+    asset,
+    btsieve::bitcoin::{BitcoindConnector, Cache},
+    htlc_location, identity,
+    swap_protocols::{
+        hbit::{
+            events::{
+                Deployed, Funded, HtlcDeployed, HtlcFunded, HtlcRedeemed, HtlcRefunded, Redeemed,
+                Refunded,
+            },
+            LedgerStates,
+        },
+        ledger, state, Ledger, LocalSwapId, Role, SecretHash,
+    },
+    timestamp::Timestamp,
+    transaction,
+};
+use ::bitcoin::{
+    hashes::{hash160, Hash},
+    Address,
+};
+use blockchain_contracts::bitcoin::rfc003::bitcoin_htlc::BitcoinHtlc;
+use chrono::{NaiveDateTime, Utc};
+use futures::future::{self, Either};
+use genawaiter::{
+    sync::{Co, Gen},
+    GeneratorState,
+};
+use std::sync::Arc;
+use tracing_futures::Instrument;
+
+/// Htlc Bitcoin atomic swap protocol.
+
+/// Data required to create a swap that involves Ether.
+#[derive(Clone, Debug)]
+pub struct CreatedSwap {
+    pub amount: asset::Bitcoin,
+    pub identity: identity::Bitcoin,
+    pub network: bitcoin::Network,
+    pub absolute_expiry: u32,
+}
+
+/// Han specific data for an in progress swap.
+#[derive(Debug, Clone)]
+pub struct InProgressSwap {
+    pub ledger: Ledger,
+    pub asset: asset::Bitcoin,
+    pub refund_identity: identity::Bitcoin,
+    pub redeem_identity: identity::Bitcoin,
+    pub expiry: Timestamp, // This is the absolute_expiry for now.
+}
+
+pub async fn new_hbit_swap<L>(
+    swap_id: LocalSwapId,
+    connector: Arc<Cache<BitcoindConnector>>,
+    ledger_states: Arc<LedgerStates>,
+    htlc_params: HtlcParams<L, asset::Bitcoin, identity::Bitcoin>,
+    role: Role,
+) where
+    L: Clone + ledger::bitcoin::Bitcoin + ledger::bitcoin::Network,
+{
+    create_watcher::<_, _, _, _, htlc_location::Bitcoin, _, transaction::Bitcoin>(
+        connector.as_ref(),
+        ledger_states,
+        swap_id,
+        htlc_params,
+        Utc::now().naive_local(),
+    )
+    .instrument(tracing::error_span!("alpha_ledger", swap_id = %swap_id, role = %role))
```

Comment:
Saw the same thing in Daniel's PR. I don''t know where this was copied from but at the time when I originally wrote these tracing statements, they made sense and this `alpha_ledger` was actually always true. This is no longer the case now.

As this needs some more thinking and probably refactoring to balance duplication and correctness, I would suggest to remove it for now.

Reply:
Good catch, thanks.  I cut and pasted with no thought for the tracing - that is a habit I need to remedy :)